src/core/main.c

```diff
@@ -1585,8 +1585,9 @@ int main(int argc, char *argv[]) {
 
         /* Initialize default unit */
         if (!arg_default_unit) {
-                r = free_and_strdup(&arg_default_unit, SPECIAL_DEFAULT_TARGET);
-                if (r < 0) {
+                arg_default_unit = strdup(SPECIAL_DEFAULT_TARGET);
+                if (!arg_default_unit) {
+                        r = log_oom();
                         log_emergency_errno(r, "Failed to set default unit %s: %m", SPECIAL_DEFAULT_TARGET);
```

Comment:
now we log twice about this error... one with log_oom() on LOG_ERR level, once much more verbosely on LOG_EMERGENCY. Pick one...

Usually, when it comes to OOM we don't bother with verbose error messages and just say "Out of memory", as this is an error that usually has little to do with the specific operatoin we did but much more globally with memory presure